pytorch_blade/src/compiler/backends/engine_interface.cpp

```diff
@@ -0,0 +1,113 @@
+// Copyright 2022 The BladeDISC Authors. All rights reserved.
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "engine_interface.h"
+
+#include <torch/script.h>
+#include "common_utils/logging.h"
+
+namespace torch {
+namespace blade {
+namespace backends {
+
+namespace {
+struct EngineCreatorRegistry {
+  std::unordered_map<std::string, EngineCreatorRegister::EngineCreator> lut;
+
+  EngineCreatorRegistry& RegisterBackend(
+      const std::string& name,
+      const EngineCreatorRegister::EngineCreator& creator) {
+    auto found = lut.find(name);
+    if (found != lut.end()) {
+      LOG(WARNING) << "Overriding already registered EngineCreator for backend "
```

Comment:
what if registry multiple backends? If that causes unpredictable behavior, I prefer using `LOG(FATAL)`.

Reply:
An elder registered one would be eliminated. I think It's good to use `LOG(FATAL)`.